javascript-carousel/main.js

```diff
@@ -0,0 +1,132 @@
+const $progArea = document.querySelector('.progress-area');
+const $caroImg = document.querySelector('.carousel-img');
+const $leftArrow = document.querySelector('i[class="arrow hoverable fas fa-arrow-circle-left"]');
+const $rightArrow = document.querySelector('i[class="arrow hoverable fas fa-arrow-circle-right"]');
+let intervalNum = 0;
+
+const imageSet = [
+  {
+    itemNum: 0,
+    path: 'images/001.png',
+    altText: 'bulbasaur'
+  },
+  {
+    itemNum: 1,
+    path: 'images/004.png',
+    altText: 'charmander'
+  },
+  {
+    itemNum: 2,
+    path: 'images/007.png',
+    altText: 'squirtle'
+  },
+  {
+    itemNum: 3,
+    path: 'images/025.png',
+    altText: 'pikachu'
+  },
+  {
+    itemNum: 4,
+    path: 'images/039.png',
+    altText: 'jigglypuff'
+  }
+];
+
+intervalNum = window.setInterval(function () {
+  let currentPos = $caroImg.getAttribute('data-num');
+  currentPos = parseInt(currentPos);
+  if (currentPos === (imageSet.length - 1)) {
+    $iconList[0].setAttribute('class', 'progress-icon hoverable fas fa-gem');
+    $caroImg.setAttribute('src', imageSet[0].path);
+    $caroImg.setAttribute('alt', imageSet[0].altText);
+    $iconList[currentPos].setAttribute('class', 'progress-icon hoverable far fa-gem');
+    $caroImg.setAttribute('data-num', 0);
+  } else {
+    advanceImg();
+  }
+}, 3 * 1000);
+
+function delayTimer() {
+  intervalNum = window.setInterval(function () {
+    let currentPos = $caroImg.getAttribute('data-num');
+    currentPos = parseInt(currentPos);
+    if (currentPos === (imageSet.length - 1)) {
+      $iconList[0].setAttribute('class', 'progress-icon hoverable fas fa-gem');
+      $caroImg.setAttribute('src', imageSet[0].path);
+      $caroImg.setAttribute('alt', imageSet[0].altText);
+      $iconList[currentPos].setAttribute('class', 'progress-icon hoverable far fa-gem');
+      $caroImg.setAttribute('data-num', 0);
+    } else {
+      advanceImg();
+    }
+  }, 3 * 1000);
+}
```

Comment:
Do you need really need two chunks of code that start an interval that does the same thing?

Reply:
fixed: just called the function on load instead.